src/Components/WebAssembly/WebAssembly.Authentication/src/Services/ExpiredTokenException.cs

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Components.WebAssembly.Authentication
+{
+    /// <summary>
+    /// An <see cref="Exception"/> that is thrown when an <see cref="AuthorizationMessageHandler"/> instance
+    /// is not able to provision an access token.
+    /// </summary>
+    public class AccessTokenNotAvailableException : Exception
+    {
+        private readonly NavigationManager _navigation;
+        private readonly AccessTokenResult _tokenResult;
+
+        public AccessTokenNotAvailableException(AccessTokenResult tokenResult, NavigationManager navigation)
+            : base(message: "Unable to provision an access token for the requested scopes.")
```

Comment:
For debugging, is it worth specifying the requested scopes in this message?

Reply:
done